src/Cli/dotnet/commands/dotnet-workload/list/VisualStudioWorkloads.cs

```diff
@@ -105,10 +146,9 @@ internal static void GetInstalledWorkloads(IWorkloadResolver workloadResolver,
                         continue;
                     }
 
-                    if (visualStudioWorkloadIds.Contains(packageId, StringComparer.OrdinalIgnoreCase))
+                    if (visualStudioWorkloadIds.TryGetValue(packageId, out string workloadId))
                     {
-                        // Normalize back to an SDK style workload ID.
-                        installedWorkloadComponents.Add(packageId.Replace('.', '-'));
+                        installedWorkloadComponents.Add(workloadId);
```

Comment:
Don't you need to normalize this back still?

Reply:
Oh, that's because you're using the workloadId and not the packageId. The packageId had to be normalized but the workloadId should be fine (and should be the short version of the id, correct?)

Reply:
correct, that's why we have a dictionary that uses the VS ID as the key and the workload ID as the value because translating back from the VS ID to the workload ID is more difficult.